sklearn/metrics/tests/test_pairwise.py

```diff
@@ -103,6 +108,19 @@ def test_pairwise_distances():
     S2 = haversine_distances(X, Y)
     assert_array_almost_equal(S, S2)
 
+    # test paired haversine
+    X = rng.randn(5, 3)
+    Y = rng.randn(5, 3)
+    with pytest.raises(ValueError):
+        paired_haversine_distances(X, Y)
+
+    # compare paired to pairwise implementations
+    X = rng.randn(5, 2)
+    Y = np.tile(X[0, :], (5, 1))
+    S = haversine_distances(X, Y[0].reshape(1, -1)).ravel()
+    S2 = paired_haversine_distances(X, Y).ravel()
+    assert_array_almost_equal(S, S2)
```

Comment:
This test should test the function pairwise_distances with parameter haversine

Reply:
You see that no paired_xxx function are tested in this test function. What I would suggest is that you don’t pop the haversine function from the global but amend the test so that you only have 2 features if haversine is used. You could for example just take the first two columns so haversine is tested like all others.

Reply:
D'uh. That's an obvious mistake.

Reply:
If I don't pop it, I get a bunch of errors like this:

```        for metric in PAIRED_DISTANCES.keys():
            # Compare our (structured) implementation to scipy
            clustering = AgglomerativeClustering(
                n_clusters=10,
                connectivity=np.ones((n_samples, n_samples)),
                metric=metric,
                linkage="complete",
            )```

Should I manually exempt all of them? I.e.,

        for metric in PAIRED_DISTANCES.keys():
            if metric == 'paired_haversine':
                continue
            # Compare our (structured) implementation to scipy
            clustering = AgglomerativeClustering(
                n_clusters=10,
                connectivity=np.ones((n_samples, n_samples)),
                metric=metric,
                linkage="complete",
            )